### PR TITLE
{vis} [intel-2015a] Mesa 10.5.5 (REVIEW) 

### PR DESCRIPTION
**Pull request by hajgato**
Needs ~~#1578~~, ~~#1620~~, ~~#1989~~

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by boegel** [comment]
@hajgato: are you going to sync this up with #1226 in terms of included dependencies?

**Comment by hajgato** [comment]
@boegel: yes. I will bump the version of  deps and buiddeps. Additionally I need gallium-llvm so llvm core will go in as dep plus some configopts updated according to serve gallium-llvm  (preparing for paraview with pserver)

**Comment by boegel** [comment]
@hajgato: try and limit the size of this PR; if it goes well beyond 5 easyconfigs, split it up logically somehow into more PRs, and specify in which order they should be tackled (otherwise this beast will never make it in)

**Comment by boegel** [comment]
Jenkins: test this please

File: easybuild/easyconfigs/m/Mesa/Mesa-10.5.5-intel-2015b-Python-2.7.9.eb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+name = 'Mesa'
+version = '10.5.5'
+
+homepage = 'http://www.mesa3d.org/'
+description = """Mesa is an open-source implementation of the OpenGL specification -
+ a system for rendering interactive 3D graphics."""
+
+
+toolchain = {'name': 'intel', 'version': '2015a'}
```
Comment (wpoely86) [inline easybuild/easyconfigs/m/Mesa/Mesa-10.5.5-intel-2015b-Python-2.7.9.eb]: Rename this file to `intel-2015a` (instead of b).

Comment (boegel) [inline easybuild/easyconfigs/m/Mesa/Mesa-10.5.5-intel-2015b-Python-2.7.9.eb]: I'm on this already, working on a PR to this branch.

**Comment by ocaisa** [comment]
Test report by @ocaisa
**FAILED**
Build succeeded for 35 out of 37 (10 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2650 0 @ 2.00GHz, Python 2.7.9
See https://gist.github.com/e277395b912eddf1dc59 for a full test report.

**Comment by hajgato** [comment]
@ocaisa noveau deleted

**Comment by ocaisa** [comment]
Test report by @ocaisa
**FAILED**
Build succeeded for 9 out of 10 (10 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2650 0 @ 2.00GHz, Python 2.7.9
See https://gist.github.com/a1c11b1d1dc4177740fd for a full test report.

**Comment by hajgato** [comment]
@boegel @ocaisa I guess, we have to roll back to LLVM 3.6?

**Comment by hajgato** [comment]
Needs PR #1989

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
@hajgato: some conflicts to resolve (I missed stuff, it seems):

```
Conflict found for dependencies of Mesa-10.5.5-intel-2015a-Python-2.7.10: M4-1.4.17-GCC-4.9.2 vs M4-1.4.17-intel-2015a
Conflict found for dependencies of Mesa-10.5.5-intel-2015a-Python-2.7.10: Python-2.7.10-intel-2015a vs Python-2.7.9-intel-2015a
Conflict found for dependencies of Mesa-10.5.5-intel-2015a-Python-2.7.10: SQLite-3.8.10.2-intel-2015a vs SQLite-3.8.8.1-intel-2015a
Conflict found for dependencies of Mesa-10.5.5-intel-2015a-Python-2.7.10: Tcl-8.6.3-intel-2015a vs Tcl-8.6.4-intel-2015a
Conflict found for dependencies of Mesa-10.5.5-intel-2015a-Python-2.7.10: Tk-8.6.3-intel-2015a-no-X11 vs Tk-8.6.4-intel-2015a-no-X11
Conflict found for dependencies of Mesa-10.5.5-intel-2015a-Python-2.7.10: libX11-1.6.3-intel-2015a-Python-2.7.10 vs libX11-1.6.3-intel-2015a-Python-2.7.9
Conflict found for dependencies of Mesa-10.5.5-intel-2015a-Python-2.7.10: libxcb-1.11-intel-2015a-Python-2.7.10 vs libxcb-1.11-intel-2015a-Python-2.7.9
Conflict found for dependencies of Mesa-10.5.5-intel-2015a-Python-2.7.10: xcb-proto-1.11-intel-2015a-Python-2.7.10 vs xcb-proto-1.11-intel-2015a-Python-2.7.9
```

**Comment by hajgato** [comment]
@boegel Confilcts hopefully solved (libXext-1.3.3-intel-2015a-Python-2.7.10.eb)

**Comment by ocaisa** [comment]
Test report by @ocaisa
**SUCCESS**
Build succeeded for 14 out of 14 (11 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2650 0 @ 2.00GHz, Python 2.7.9
See https://gist.github.com/031a7c9815be9ecf43d9 for a full test report.

**Comment by ocaisa** [comment]
I don't really like having the explicit dependencies on software built with `GCC/4.9.2` when it's not really necessary (this can cause problems in a hierarchical MNS):
```
libpciaccess-0.13.1-intel-2015a.eb:    ('Autotools', '20150119', '', ('GCC', '4.9.2')),
Mesa-10.5.5-intel-2015a-Python-2.7.10.eb:    ('flex', '2.5.39', '', ('GCC', '4.9.2')),
Mesa-10.5.5-intel-2015a-Python-2.7.10.eb:    ('Bison', '3.0.4', '', ('GCC', '4.9.2')),
Mesa-10.5.5-intel-2015a-Python-2.7.10.eb:    ('Autotools', '20150119', '', ('GCC', '4.9.2')),
Mesa-10.5.5-intel-2015a-Python-2.7.10.eb:    ('pkg-config', '0.28', '', ('GCC', '4.9.2')),
Mesa-10.5.5-intel-2015a-Python-2.7.10.eb:    ('M4', '1.4.17', '', ('GCC', '4.9.2')),
xorg-macros-1.17-intel-2015a.eb:dependencies = [('Autotools', '20150119', '', ('GCC', '4.9.2'))]
```
I'm just testing now if this can be successfully avoided without Autotools, M4 etc. built with `intel/2015a` toolchain

**Comment by boegel** [comment]
@ocaisa: once your subtoolchain PR is merged, we can simply trash these subtoolchain specs?

**Comment by ocaisa** [comment]
@hajgato I managed to build independent of GCC working hajgato/easybuild-easyconfigs#28

**Comment by boegel** [comment]
@ocaisa: I doubt the GMP one with `intel/2015b` works, did you test it?

**Comment by ocaisa** [comment]
Yup, we have it installed with the same compiler, the crucial thing is the precise compiler flag...but I'll check it again now just in case.

**Comment by boegel** [comment]
@ocaisa: ah, interesting!

A bit ironic that you're now trying to get people to use full toolchains btw ;)

**Comment by ocaisa** [comment]
Well, I'm preparing for the day when this doesn't matter any more...a day that I am making happen! Having to explicitly edit this stuff for updates is exceedingly painful

**Comment by boegel** [comment]
:+1:

**Comment by hajgato** [comment]
@ocaisa And does it work on Haswell architecture as well? Because GMP was problematic on Haswell with Intel compilers, but worked on older architectures.

**Comment by hajgato** [comment]
@ocasia Still some conflicts. See https://jenkins1.ugent.be/job/easybuild-easyconfigs-pr-builder/4560/console for more details.

**Comment by ocaisa** [comment]
Conflicts fixed in hajgato/easybuild-easyconfigs#30 but I needed to bring the branch up to date with devel too for that. Maybe this is getting too messy? Should we proceed with this merge and I make a separate PR to make it avoid cross toolchain dependencies.

**Comment by wpoely86** [comment]
@ocaisa no, a sync with develop is always ugly as a PR. If we just change the CGAL conflicts in this PR, it's fine. Once @hajgato merges it, it will look much nicer here.

**Comment by hajgato** [comment]
@ocaisa I agree, this goes too far. I think we have to merge mesa, and then fix GMP/intel issues in a separate PR. @boegel what is your opinion?

**Comment by boegel** [comment]
Basically, the only thing that changed in @ocaisa's PR is to use `intel/2015a` for GMP rather than `GCC/4.9.2`?

In that case, this PR can go in, and the change that @ocaisa did for the toolchain used for GMP can be done in a follow-up PR, sure.

I want to avoid merging things in develop that don't work, but that's not the case here, as far as I understand?

**Comment by ocaisa** [comment]
I'm fine with that, I have a separate PR ready and waiting for this to go in. I verified the original PR already and it built ok for me.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 0 out of 0 (11 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/e050eba3d0473a04a635 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 11 out of 11 (11 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/a0546a6eb7879ed86472 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 31 out of 31 (11 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/1e03c77572ec177c62cf for a full test report.

**Comment by boegel** [comment]
OK, (finally) going in, thanks @hajgato and everyone else involved with this one!

**Pull request merged**